dashboard/app/models/user.rb

```diff
@@ -246,6 +247,24 @@ def delete_course_as_facilitator(course)
     courses_as_facilitator.find_by(course: course).try(:destroy)
   end
 
+  def log_admin_save
+    # Do not log for development, test, and adhoc environments.
+    return unless should_log?
+
+    ChatClient.message 'infra-security',
+      "#{admin ? 'Granting' : 'Revoking'} UserPermission: "\
+      "environment: #{rack_env}, "\
+      "user ID: #{id}, "\
+      "email: #{email}, "\
+      "permission: ADMIN",
+      color: 'yellow'
+  end
+
+  # don't log changes to admin permission in development, test, and ad_hoc environments
+  def should_log?
```

Comment:
The name of this method doesn't seem right to me. If I see `some_user.should_log?`, I don't expect it to simply depend on the rack environment.

Q: Should this be `def self.should_log?`?
Q: Should this be `def self.should_log_to_chat_client?`
Q: Should this logic be moved from `log_admin_save` to the `before_save` callback calling that method? E.g., something like `if: :admin_changed? && [:staging, :levelbuilder, :production].include? rack_env`.